cachi2/interface/cli.py

```diff
@@ -97,14 +96,58 @@ def _looks_like_json(value: str) -> bool:
     return value.lstrip().startswith(("{", "["))
 
 
+class _Input:
+    """
+    Parses the input argument in the fetch-deps function.
+
+    Since this argument can take multiple forms, this parsing helps the convertion to
+    actual input models.
+
+    See the fetch-deps docs for examples of valid inputs.
+    """
+
+    packages: list[dict]
+    flags: list[str] = list()
+
+    def __init__(self, input: str) -> None:
+        if _looks_like_json(input):
+            self._parse_json(input)
+        else:
+            self._parse_packages(input)
+
+    def _parse_json(self, input: str) -> None:
+        input_obj = json.loads(input)
+
+        if isinstance(input_obj, dict) and "packages" in input_obj.keys():
+            self.packages = input_obj["packages"]
+            self._parse_flags(input_obj.get("flags", []))
+            return
+
+        self._parse_packages(input_obj)
+
+    def _parse_packages(self, packages: Any) -> None:
+        if isinstance(packages, list):
+            self.packages = packages
+        elif isinstance(packages, dict):
+            self.packages = [packages]
+        else:
+            self.packages = [{"type": packages, "path": "."}]
+
+    def _parse_flags(self, flags: Any):
+        if isinstance(flags, list):
+            self.flags = [str(flag) for flag in flags]
+        else:
+            self.flags = [str(flags)]
```

Comment:
Should we allow only lists of flags, consistently with `"packages"`?

Reply:
Yes, I changed it so it'll only accept a list now.